### PR TITLE
Minor diff reduction between RLOO and GRPO

### PR DESCRIPTION
**Pull request by qgallouedec**
While checking the diff between RLOO and GRPO, I realized some useless line diff, this PR aims to reduce them

---

> [!NOTE]
> **Low Risk**
> Mostly formatting/logging cleanups; the only behavioral changes are using numeric `global_step` in W&B tables and switching to in-place temperature scaling (`div_`) which should be equivalent but could affect tensor aliasing in edge cases.
> 
> **Overview**
> Reduces diff noise across GRPO-related trainers by tightening formatting (imports, padding calls, and docstring citation blocks) and adding a clarifying comment in `RLOOTrainer`.
> 
> Also adjusts completion logging tables in `GOLDTrainer` and `RLOOTrainer` to log `step` as an integer `global_step` rather than a string, and uses in-place `logits.div_(temperature)` for temperature scaling to match style and reduce churn.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 7d5a6af892358e78c428c769e52f2dd3615ae6f7. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: trl/experimental/gold/gold_trainer.py [hunk 1]
```diff
@@ -2402,7 +2402,7 @@ def log(self, logs: dict[str, float], start_time: float | None = None) -> None:
                 import pandas as pd
 
                 table = {
-                    "step": [str(self.state.global_step)] * len(self._textual_logs["prompt"]),
+                    "step": [self.state.global_step] * len(self._textual_logs["prompt"]),
```
Comment (qgallouedec) [inline trl/experimental/gold/gold_trainer.py:2405]: comes from here https://github.com/huggingface/trl/pull/4818/changes#diff-964e6fd373aa93037604064cb2b822d7f8e2735e33f791065acf2c4c3552d393R2169

File: trl/trainer/rloo_trainer.py [hunk 2]
```diff
@@ -726,7 +726,7 @@ def _get_per_token_logps_and_entropies(
             logits = logits[:, -logits_to_keep:, :]  # (B, logits_to_keep, H)
             # Divide logits by sampling temperature.
             # See https://huggingface.co/blog/the_n_implementation_details_of_rlhf_with_ppo#policy-training-implementation-details
-            logits = logits / self.temperature
+            logits.div_(self.temperature)
```
Comment (qgallouedec) [inline trl/trainer/rloo_trainer.py:729]: comes from https://github.com/huggingface/trl/pull/5254
